src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -338,17 +337,23 @@ private void AddNameAndTypeArgumentsOrParameters(INamedTypeSymbol symbol)
 
             var partKind = GetPartKind(symbol);
 
-            if (symbolName == null)
-            {
-                symbolName = symbol.Name;
-            }
+            symbolName ??= symbol.Name;
+
+            var renderSimpleAnonymousDelegate =
+                symbol.TypeKind == TypeKind.Delegate &&
+                !symbol.CanBeReferencedByName &&
+                !format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.UseMetadataMethodNames);
 
             if (format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.UseErrorTypeSymbolName) &&
                 partKind == SymbolDisplayPartKind.ErrorTypeName &&
                 string.IsNullOrEmpty(symbolName))
             {
                 builder.Add(CreatePart(partKind, symbol, "?"));
             }
+            else if (renderSimpleAnonymousDelegate)
+            {
+                builder.Add(new SymbolDisplayPart(SymbolDisplayPartKind.DelegateName, null, "<anonymous delegate>"));
```

Comment:
Are these things that we try to localize? I'm not familiar with our policy on that for this type.

Reply:
we don't do it elsewhere in this type for things like anonymous-types.  so i'm keeping with that pattern.